From a5d5a36f80d5e37a8f06eb53d8bbedcf9fc6f01c Mon Sep 17 00:00:00 2001
From: Owen Huang <Owen.Huang@infineon.com>
Date: Tue, 12 Jul 2022 01:56:58 -0500
Subject: [PATCH 130/137] non-upstream: Fix kernel crash caused by race on
 timer_add

brcmf_sdio_wd_timer must be protected using the
sdio_claim_host and sdio_release_host


Signed-off-by: Owen Huang <Owen.Huang@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
index bc27bc7e3d9a..c302ef680cd0 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
@@ -4764,11 +4764,10 @@ static void brcmf_sdio_firmware_callback(struct device *dev, int err,
 		goto checkdied;
 	}
 
-	sdio_release_host(sdiod->func1);
-
 	/* Start the watchdog timer */
 	bus->sdcnt.tickcnt = 0;
 	brcmf_sdio_wd_timer(bus, true);
+	sdio_release_host(sdiod->func1);
 
 	err = brcmf_alloc(sdiod->dev, sdiod->settings);
 	if (err) {
-- 
2.17.1

